### PR TITLE
Add length validation to add Image endpoint and Test

### PR DESCRIPTION
**Pull request by sakataalan**

### CODE REVIEW

**Comment by sakataalan** [comment]
[Add length validation to add Image endpoint](https://trello.com/c/HtqL7OqR/12-add-length-validation-to-add-image-endpoint)

File: src/main/kotlin/com/game/catbackend/domain/services/ImageService.kt [hunk 1]
```diff
@@ -14,6 +14,9 @@ class ImageService(val imageRepository: ImageRepository) {
         val image = imageDto.toImage()
         image.active = true
 
+        if (image.source.length > 200) {
+            return throw Exception("The len of image source needs to be less than 200 characters!")
```
Comment (Orlando-Silva) [inline src/main/kotlin/com/game/catbackend/domain/services/ImageService.kt]: Create a more especific exception for this error. Something among the lines of ImageInvalidAttributeException

File: src/test/kotlin/com/game/catbackend/domain/services/ImageServiceTests.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+
+class ImageServiceTests() : CatBackendBaseTest(){
+
+    @Autowired
+    lateinit var imageService: ImageService
+
+    @Test
+     fun `when add is called with source length more than 200 characters it should return an exception`() {
```
Comment (Orlando-Silva) [inline src/test/kotlin/com/game/catbackend/domain/services/ImageServiceTests.kt:18]: 🪨

File: src/test/kotlin/com/game/catbackend/domain/services/ImageServiceTests.kt [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        val source = "https://i.natgeofe.com/n/548467d8-c5f1-4551-9f58-6817a8d2c45e/NationalGeographic_2572187_square/NationalGeographic_2572187_square/NationalGeographic_2572187_square/NationalGeographic_2572187_square/NationalGeographic_2572187_square.jpg";
+
+        val imageErrorException = catchThrowable {
+            imageService.add(ImageDTO(source))
+        }
+
+        assertThat(imageErrorException).isNotNull()
```
Comment (Orlando-Silva) [inline src/test/kotlin/com/game/catbackend/domain/services/ImageServiceTests.kt]: Here you can also check the exception class

File: pom.xml [hunk 4]
```diff
@@ -79,7 +79,17 @@
 			<version>5.9.2</version>
 			<scope>test</scope>
 		</dependency>
-	</dependencies>
+		<dependency>
```
Comment (Orlando-Silva) [inline pom.xml:82]: Remove this

File: src/main/kotlin/com/game/catbackend/CatBackendApplication.kt [hunk 5]
```diff
@@ -2,6 +2,7 @@ package com.game.catbackend
 
 import org.springframework.boot.autoconfigure.SpringBootApplication
 import org.springframework.boot.runApplication
+import org.springframework.context.annotation.Import
```
Comment (Orlando-Silva) [inline src/main/kotlin/com/game/catbackend/CatBackendApplication.kt]: Remove this

File: src/main/kotlin/com/game/catbackend/api/ImageAPI.kt [hunk 6]
```diff
@@ -3,6 +3,8 @@ package com.game.catbackend.api
 import com.game.catbackend.api.dto.ImageDTO
 import com.game.catbackend.domain.entities.Image
 import com.game.catbackend.domain.services.ImageService
+import jakarta.validation.Valid
+import org.springframework.validation.BindingResult
```
Comment (Orlando-Silva) [inline src/main/kotlin/com/game/catbackend/api/ImageAPI.kt]: Remove this

File: src/main/kotlin/com/game/catbackend/api/dto/ImageDTO.kt [hunk 7]
```diff
@@ -1,9 +1,11 @@
 package com.game.catbackend.api.dto
 
 import com.game.catbackend.domain.entities.Image
+import jakarta.validation.constraints.Size
 
-data class ImageDTO(
-    val source: String
+class ImageDTO(
+    @Size(max = 200, message = "The len of image source needs to be less than 200 characters!")
```
Comment (Orlando-Silva) [inline src/main/kotlin/com/game/catbackend/api/dto/ImageDTO.kt]: The *length*

File: src/main/kotlin/com/game/catbackend/domain/entities/Image.kt [hunk 8]
```diff
@@ -1,6 +1,7 @@
 package com.game.catbackend.domain.entities
 
 import jakarta.persistence.*
+import jakarta.validation.constraints.Size
```
Comment (Orlando-Silva) [inline src/main/kotlin/com/game/catbackend/domain/entities/Image.kt]: Remove this

File: src/main/kotlin/com/game/catbackend/domain/services/ImageService.kt [hunk 9]
```diff
@@ -3,8 +3,11 @@ package com.game.catbackend.domain.services
 import com.game.catbackend.api.dto.ImageDTO
 import com.game.catbackend.api.dto.toImage
 import com.game.catbackend.domain.entities.Image
+import com.game.catbackend.domain.exceptions.CatGameBaseException
 import com.game.catbackend.infra.repositories.ImageRepository
+import jakarta.validation.Valid
```
Comment (Orlando-Silva) [inline src/main/kotlin/com/game/catbackend/domain/services/ImageService.kt]: Remove this

File: src/main/kotlin/com/game/catbackend/domain/services/ImageService.kt [hunk 10]
```diff
@@ -3,8 +3,11 @@ package com.game.catbackend.domain.services
...
 import org.springframework.stereotype.Service
+import org.springframework.validation.annotation.Validated
```
Comment (Orlando-Silva) [inline src/main/kotlin/com/game/catbackend/domain/services/ImageService.kt]: Remove this

File: src/main/kotlin/com/game/catbackend/domain/services/ImageService.kt [hunk 11]
```diff
@@ -14,6 +17,9 @@ class ImageService(val imageRepository: ImageRepository) {
         val image = imageDto.toImage()
         image.active = true
 
+        if (image.source.length > 200) {
+            return throw CatGameBaseException("The len of image source needs to be less than 200 characters!")
```
Comment (Orlando-Silva) [inline src/main/kotlin/com/game/catbackend/domain/services/ImageService.kt]: CatGameInvalidAttributeException

**Review by Orlando-Silva — APPROVED** [review]

**Pull request merged**